Fix tests for onnx 1.13

ONNX 1.13 produced some errors in tests: https://github.com/microsoft/onnx-script/actions/runs/3678076190/jobs/6220897226

```
E           onnx.onnx_cpp2py_export.checker.ValidationError: Field 'type' of 'value_info' is required but missing.
```

--- a/.github/workflows/lint.yaml
+++ b/.github/workflows/lint.yaml
@@ -52,7 +52,7 @@ jobs:
           # Install ONNX
           python -m pip uninstall -y onnx-function-experiment
           python -m pip uninstall -y ort-function-experiment-nightly
-          python -m pip install onnx onnxruntime
+          python -m pip install -r requirements-onnx.txt
           # Install packages
           python -m pip install -e .
           lintrunner init

--- a/.github/workflows/lint.yaml
+++ b/.github/workflows/lint.yaml
@@ -53,7 +53,7 @@ jobs:
           python -m pip uninstall -y onnx-function-experiment
           python -m pip uninstall -y ort-function-experiment-nightly
           python -m pip install onnx onnxruntime
-          # install packages
+          # Install packages
           python -m pip install -e .
           lintrunner init
       - name: Run lintrunner on all files

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -63,12 +63,13 @@ jobs:
           python -m pip install --upgrade pip setuptools wheel build
           python -m pip install -r requirements-dev.txt
 
+      # TODO(#249): Fix tests for onnx 1.13
       - name: Install standard onnx
         if: ${{ matrix.onnx_standard }}
         run: |
           python -m pip uninstall -y onnx-function-experiment
           python -m pip uninstall -y ort-function-experiment-nightly
-          python -m pip install onnx onnxruntime
+          python -m pip install onnx==1.12 onnxruntime
 
       - name: Versions
         run: |

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -63,13 +63,12 @@ jobs:
           python -m pip install --upgrade pip setuptools wheel build
           python -m pip install -r requirements-dev.txt
 
-      # TODO(#249): Fix tests for onnx 1.13
       - name: Install standard onnx
         if: ${{ matrix.onnx_standard }}
         run: |
           python -m pip uninstall -y onnx-function-experiment
           python -m pip uninstall -y ort-function-experiment-nightly
-          python -m pip install onnx==1.12 onnxruntime
+          python -m pip install -r requirements-onnx.txt
 
       - name: Versions
         run: |

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -25,13 +25,12 @@ steps:
     python -m pip install -r requirements-dev.txt
   displayName: 'Install dependencies'
 
-# TODO(#249): Fix tests for onnx 1.13
 - script: |
       if [ '$(onnx.standard)' == '1' ]
         then
           python -m pip uninstall -y onnx-function-experiment
           python -m pip uninstall -y ort-function-experiment-nightly
-          python -m pip install onnx==1.12 onnxruntime
+          python -m pip install -r requirements-onnx.txt
       fi
   displayName: 'Install onnx'
 

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -25,12 +25,13 @@ steps:
     python -m pip install -r requirements-dev.txt
   displayName: 'Install dependencies'
 
+# TODO(#249): Fix tests for onnx 1.13
 - script: |
       if [ '$(onnx.standard)' == '1' ]
         then
           python -m pip uninstall -y onnx-function-experiment
           python -m pip uninstall -y ort-function-experiment-nightly
-          python -m pip install onnx onnxruntime
+          python -m pip install onnx==1.12 onnxruntime
       fi
   displayName: 'Install onnx'
 

--- a/onnxscript/backend/onnx_backend.py
+++ b/onnxscript/backend/onnx_backend.py
@@ -74,7 +74,7 @@ class OnnxBackendTest:
             seq = onnx.SequenceProto()
             try:
                 seq.ParseFromString(serialized)
-                loaded = to_list(seq)
+                loaded = to_list(seq)  # type: ignore[assignment]
             except Exception:  # pylint: disable=W0703
                 try:
                     loaded = onnx.load_model_from_string(serialized)

--- a/onnxscript/backend/onnx_export.py
+++ b/onnxscript/backend/onnx_export.py
@@ -207,7 +207,10 @@ class Exporter:
         if hasattr(graph, "initializer"):
             for init in graph.initializer:
                 node = make_node(
-                    "Constant", [], [self._rename_variable(init.name)], value=init
+                    "Constant",
+                    [],
+                    [self._rename_variable(init.name)],  # type: ignore[list-item]
+                    value=init,
                 )
                 code.append(self._python_make_node(node, opsets, indent=indent))
         if hasattr(graph, "sparse_initializer") and len(graph.sparse_initializer) > 0:

--- a/onnxscript/irbuilder.py
+++ b/onnxscript/irbuilder.py
@@ -183,10 +183,10 @@ class IRFunction:
         self.inputs: list[IRVar] = []
         self.outputs: list[IRVar] = []
         self.stmts: list[IRStmt] = []
-        self.attrs: list[str] = []  # attribute parameters
-        self.attr_protos: list[
-            onnx.AttributeProto
-        ] = []  # attribute parameters with default value
+        # attribute parameters
+        self.attrs: list[str] = []
+        # attribute parameters with default value
+        self.attr_protos: list[IRAttributeValue] = []
         self.called_functions: dict[str, onnx.FunctionProto] = {}
         self.docstring: str = ""
         # a dictionary of nested function-definitions
@@ -472,5 +472,7 @@ class IRBuilder:
         a = onnx.AttributeProto()
         a.name = attrname
         a.ref_attr_name = refname
-        a.type = ta.pytype_to_attrtype(pytype)
+        type_ = ta.pytype_to_attrtype(pytype)
+        assert type_ is not None
+        a.type = type_
         return IRAttributeValue(a)

--- a/onnxscript/test/common/onnx_script_test_case.py
+++ b/onnxscript/test/common/onnx_script_test_case.py
@@ -54,11 +54,11 @@ class OnnxScriptTestCase(unittest.TestCase):
         try:
             # experimental version
             # pylint: disable=no-value-for-parameter
-            cls.all_test_cases = node_test.collect_testcases()  # type: ignore[attr-defined]
+            cls.all_test_cases = node_test.collect_testcases()  # type: ignore[attr-defined,call-arg]
             # pylint: enable=no-value-for-parameter
         except TypeError:
             # official version
-            cls.all_test_cases = node_test.collect_testcases(None)  # type: ignore[attr-defined]
+            cls.all_test_cases = node_test.collect_testcases(None)  # type: ignore[attr-defined,arg-type]
 
     def _create_model_from_param(
         self, param: FunctionTestParams, onnx_case_model: onnx.ModelProto

--- a/onnxscript/utils.py
+++ b/onnxscript/utils.py
@@ -20,7 +20,7 @@ try:
     from onnx.printer import to_text as proto2text
 except ImportError:
 
-    def proto2text(x):  # pylint: disable=unused-argument
+    def proto2text(_) -> str:  # type: ignore[misc]
         return "<print utility unavailable>"
 
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -49,20 +49,17 @@ filterwarnings = ["ignore::UserWarning", "ignore::DeprecationWarning"]
 [tool.mypy]
 follow_imports = "silent"   # TODO: Remove when we fix all the mypy errors
 strict_optional = true
-warn_return_any = true
 warn_no_return = true
-warn_unused_ignores = true
+warn_unused_ignores = false # TODO: CI and local are inconsistent when this is true. Investigate.
 warn_redundant_casts = true
 warn_incomplete_stub = true
 # TODO disallow_untyped_calls = true
 check_untyped_defs = true
-disallow_any_generics = true
-no_implicit_optional = true
+disallow_any_generics = false
 # TODO disallow_incomplete_defs = true
 # TODO disallow_subclassing_any = true
 disallow_untyped_decorators = true
 warn_unused_configs = true
-show_error_codes = true
 show_column_numbers = true
 
 [[tool.mypy.overrides]]

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -25,7 +25,7 @@ pytest-subtests
 pytest-xdist
 pytest!=7.1.0
 pyyaml
-torch
+torch>=1.13
 
 # Lint
 lintrunner

--- a/requirements-onnx.txt
+++ b/requirements-onnx.txt
@@ -1,0 +1,5 @@
+# Requirements for testing with the release version of onnx and onnxruntime
+
+# TODO(#249): Fix tests for onnx 1.13
+onnx==1.12
+onnxruntime

